### PR TITLE
bugfix issue 3135

### PR DESCRIPTION
**Pull request by Ryanglambert**
fixes issue (includes test)

### CODE REVIEW

File: llama_index/output_parsers/selection.py [hunk 1]
```diff
@@ -45,7 +46,15 @@ class Answer(DataClassJsonMixin):
 
 
 class SelectionOutputParser(BaseOutputParser):
+    def _marshal_to_json(self, output: str) -> str:
+        "llm output to json friendly"
```
Comment (jerryjliu) [inline llama_index/output_parsers/selection.py:50]: could you make this docstring well-formatted? with three triple quotes on each side

File: llama_index/output_parsers/selection.py [hunk 2]
```diff
@@ -45,7 +46,15 @@ class Answer(DataClassJsonMixin):
...
+        output = output.strip()
+        if output.startswith("Output:"):
```
Comment (jerryjliu) [inline llama_index/output_parsers/selection.py:52]: hardcoding this is a bit hacky tbh. can we instead do `find` and `rfind` for a `{`/`[` or `}`/`]` characters respectively? that way you prune out any output before or after the actual json

File: tests/output_parsers/test_selection.py [hunk 3]
```diff
@@ -26,3 +27,19 @@ def test_parse(output_parser: SelectionOutputParser) -> None:
     assert len(parsed.parsed_output) == 2
     assert parsed.parsed_output[0].choice == 1
     assert parsed.parsed_output[0].reason == "just because"
+
+
+def test_bugfix_llm_output_with_string_outside_of_json_issue_3135(
```
Comment (jerryjliu) [inline tests/output_parsers/test_selection.py:32]: let's not name this after an issue, and more just test_selection_parse_to_dict or something equivalent

File: tests/output_parsers/test_selection.py [hunk 4]
```diff
@@ -26,3 +27,19 @@ def test_parse(output_parser: SelectionOutputParser) -> None:
...
+    output = """
+    Output:
+    [
+      {
+        "choice": 1,
+        "reason": "Useful for questions that seek to gain a "
+      }
+    ]
+    """
+    output_parser.parse(output)
```
Comment (jerryjliu) [inline tests/output_parsers/test_selection.py:45]: can we assert that this matches the expected dict?

Comment (Ryanglambert) [reply tests/output_parsers/test_selection.py]: Thanks for all the feedback!

**Review by jerryjliu — APPROVED** [review]

**Pull request merged**